sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -111,11 +111,11 @@ abstract class JdbcDialect extends Serializable with Logging {
    * JDBC dialects should override this function to provide implementations that suite their
    * JDBC drivers.
    * @param t Timestamp returned from JDBC driver getTimestamp method.
-   * @return A LocalDateTime representing the same wall clock time as the timestamp in database.
+   * @return A Long value representing the same wall clock time as the timestamp in database.
    */
   @Since("3.5.0")
-  def convertJavaTimestampToTimestampNTZ(t: Timestamp): LocalDateTime = {
-    DateTimeUtils.microsToLocalDateTime(DateTimeUtils.fromJavaTimestampNoRebase(t))
+  def convertJavaTimestampToTimestampNTZ(t: Timestamp): Long = {
```

Comment:
I think symmetry API is more important than tiny perf improvements. But if you have benchmark numbers to show this is a big issue, I'm happy to change my mind.

Reply:
Yes. the performance is very bad.